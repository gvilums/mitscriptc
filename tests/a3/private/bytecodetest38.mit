// let's make a big cycle
start = {};
current = start;
x = 0;

while (x < 100) {
    current.x = x;
    current.next = {};
    current.next.previous = current;
    current = current.next;

    x = x + 1;
}

current = current.previous;

current.next = start;
start.previous = current;

print(current.previous.previous.x);
print(current.previous.x);
print(current.x);
print(current.next.x);
print(current.next.next.x);
