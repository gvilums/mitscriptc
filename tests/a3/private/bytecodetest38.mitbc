function
{
	functions =
	[
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 0,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		},
		function
		{
			functions = [],
			constants = [],
			parameter_count = 1,
			local_vars = [],
			local_ref_vars = [],
			free_vars = [],
			names = [],
			instructions = 
			[
			]
		}
	],
	constants = [0, 100, 1],
	parameter_count = 0,
	local_vars = [],
	local_ref_vars = [],
	free_vars = [],
	names = [start, current, x, next, previous, print, input, intcast],
	instructions = 
	[
		load_func	0
		alloc_closure	0
		store_global	5
		load_func	1
		alloc_closure	0
		store_global	6
		load_func	2
		alloc_closure	0
		store_global	7
		alloc_record
		store_global	0
		load_global	0
		store_global	1
		load_const	0
		store_global	2
		load_global	2
		load_const	1
		swap
		gt
		not
		if	19
		load_global	1
		load_global	2
		field_store	2
		load_global	1
		alloc_record
		field_store	3
		load_global	1
		field_load	3
		load_global	1
		field_store	4
		load_global	1
		field_load	3
		store_global	1
		load_global	2
		load_const	2
		add
		store_global	2
		goto	-23
		load_global	1
		field_load	4
		store_global	1
		load_global	1
		load_global	0
		field_store	3
		load_global	0
		load_global	1
		field_store	4
		load_global	5
		load_global	1
		field_load	4
		field_load	4
		field_load	2
		call	1
		pop
		load_global	5
		load_global	1
		field_load	4
		field_load	2
		call	1
		pop
		load_global	5
		load_global	1
		field_load	2
		call	1
		pop
		load_global	5
		load_global	1
		field_load	3
		field_load	2
		call	1
		pop
		load_global	5
		load_global	1
		field_load	3
		field_load	3
		field_load	2
		call	1
		pop
	]
}