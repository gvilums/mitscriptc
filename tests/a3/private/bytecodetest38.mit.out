97
98
99
0
1
